crates/stc_ts_file_analyzer/src/analyzer/expr/mod.rs

```diff
@@ -3147,8 +3146,34 @@ impl Analyzer<'_, '_> {
     #[cfg_attr(debug_assertions, tracing::instrument(skip_all))]
     pub(crate) fn expand_generics_with_type_args(&mut self, span: Span, ty: Type, type_args: &TypeParamInstantiation) -> VResult<Type> {
         match ty.normalize() {
-            Type::Interface(Interface { type_params, .. })
-            | Type::Alias(Alias { type_params, .. })
+            Type::Interface(Interface { type_params, body, .. }) => {
+                let mut params = HashMap::default();
+                let mut type_params_vec: Vec<TypeParam> = vec![];
```

Comment:
Why do we need this?

Reply:
```ts
interface BConstructor {
  new <T>(): B<T>;
}
interface B<T> {
  foo: T;
}
declare var B: BConstructor;

var obj4: any;
if (obj4 instanceof B) {
  obj4; // B<any>
  obj4.foo = "str";
  obj4.foo = 1;
}
```

Reply:
BConstructor has not type_params, but constructor has type_param

Reply:
I mean, instead of 

```rust
a.extend(b);

for (b, c) in b.iter().zip(c) {
}
```

we can do

```rust
for (b, c) in a.iter().zip(c) {
}
```

Of course, as we have two `extend`, we will need two loops.
But I think two loops version is more intuitive, as appending type parameters to a vector is something which is not intuitive at all.

Reply:
umm... this fixed it, next commit
https://github.com/dudykr/stc/pull/485/commits/00a8f87b4329ec3d64c995a3aad1aab5a642b99e#

Reply:
thx.